API/enums/calculationTypes.m

```diff
@@ -1,11 +1,15 @@
 classdef calculationTypes < customEnum
-    methods (Static)
+   methods (Static)
+        function s = toStruct()
+            s = customEnum.toStruct(mfilename('class'));
+        end
+        
         function v = values()
-            v = customEnum.values('calculationTypes');
+            v = customEnum.values(mfilename('class'));
         end
         
         function e = fromValue(value)
-            e = customEnum.fromValue('calculationTypes', value);
+            e = customEnum.fromValue(mfilename('class'), value);
         end
     end
```

Comment:
Can we define a (say) "RATEnum" with these (now identical) routines and inherit for each specific enum?

Reply:
I am not sure I fully get your suggestion but sadly mfilename will only work in the current file it will not work with inheritance or any other trick I have tried so far

Reply:
Righto, I thought that might be the problem. If you look at each of the enums the three methods are identical so it seems natural to ask about inheritance.

Reply:
Yeah, I will revisit if I find a better solution